### PR TITLE
Fix screenbuilder attempt to launch simulator regardless of OS

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
This causes weird behavior on Linux as well as under other OS in which the simulator cannot be run (for example Windows without .NET Framework)
Fixes [#297486](http://teampulse.telerik.com/view#item/297486)

### CODE REVIEW

File: lib/commands/simulate.ts [hunk 1]
```diff
@@ -13,23 +13,13 @@ export class SimulateCommand implements ICommand {
 	}
 
 	public execute(args: string[]): IFuture<void> {
-		return this.$simulatorService.launchSimulator();
+		return (() => {
+			if (this.$simulatorPlatformServices.canRunApplication().wait()) {
```
Comment (rosen-vladimirov) [inline lib/commands/simulate.ts]: simulatorPlatoformServices is {} on Linux, so this code fails with error:
```
TypeError: Object #<Object> has no method 'canRunApplication'
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:488:15)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/dispatchers.ts:46:74)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    - - - - -
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:488:15)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:59:39)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    - - - - -
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/commands/simulate.ts:17:39)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:59:39)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
```

File: lib/commands/project/create.ts [hunk 2]
```diff
@@ -45,8 +46,8 @@ export class CreateCommand extends ProjectCommandBaseLib.ProjectCommandBase {
 				this.$fs.deleteDirectory(projectPath).wait();
 				throw err;
 			}
-			
-			if (this.$options.simulator) {
+
+			if (this.$options.simulator && this.$simulatorPlatformServices.canRunApplication().wait()) {
```
Comment (rosen-vladimirov) [inline lib/commands/project/create.ts]: simulatorPlatoformServices is {} on Linux, so this code fails with error:
```
TypeError: Object #<Object> has no method 'canRunApplication'
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:488:15)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/yok.ts:239:105)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    - - - - -
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:488:15)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:59:39)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    - - - - -
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/commands/project/create.ts:50:66)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:59:39)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/errors.ts:118:20)
    at Function.settle (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:220:26)
    at FiberFuture.Future.wait (/home/rvladimirov/Work/icenium-cli/node_modules/fibers/future.js:487:10)
    at <anonymous> (/home/rvladimirov/Work/icenium-cli/lib/common/services/commands-service.ts:95:91)
```

File: lib/commands/project/create.ts [hunk 3]
```diff
@@ -45,8 +46,8 @@ export class CreateCommand extends ProjectCommandBaseLib.ProjectCommandBase {
 				this.$fs.deleteDirectory(projectPath).wait();
 				throw err;
 			}
-			
-			if (this.$options.simulator) {
+
+			if (this.$options.simulator && this.$simulatorPlatformServices && this.$simulatorPlatformServices.canRunApplication().wait()) {
```
Comment (rosen-vladimirov) [inline lib/commands/project/create.ts]: `this.$simulatorPlatformServices` is empty object, so  `if(this.$simulatorPlatformServices)` is true. You should use:
```TypeScript
if (this.$options.simulator && this.$simulatorPlatformServices.canRunApplication && this.$simulatorPlatformServices.canRunApplication().wait()) {
```

File: lib/commands/simulate.ts [hunk 4]
```diff
@@ -13,23 +13,13 @@ export class SimulateCommand implements ICommand {
 	}
 
 	public execute(args: string[]): IFuture<void> {
-		return this.$simulatorService.launchSimulator();
+		return (() => {
+			if (this.$simulatorPlatformServices && this.$simulatorPlatformServices.canRunApplication().wait()) {
```
Comment (rosen-vladimirov) [inline lib/commands/simulate.ts]: `this.$simulatorPlatformServices` is empty object, so  `if(this.$simulatorPlatformServices)` is true. You should use:
```TypeScript
if (this.$options.simulator && this.$simulatorPlatformServices.canRunApplication && this.$simulatorPlatformServices.canRunApplication().wait()) {
```

**Comment by Mitko-Kerezov** [comment]
Comments by @rosen-vladimirov addressed

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**